File: Algorithms/Sorting-Algorithms/Sorting-Problems/move_zeroes/Python/moveZeroes.py
import array

def move_zeroes_bubble_array(arr):
	#length of the array
	n = len(arr)

	for i in range(n):
		flag = False

		# Check all adjacent pairs 
		for j in range(n-1):
			# finding elements where j-->0 and j+1 --> not equals to 0
			if arr[j] == 0 and arr[j+1] != 0: 
				temp = arr[j]
				arr[j] = arr[j+1]
				arr[j+1] = temp
				flag = True

		if flag == False:
			break

	return arr


arr = array.array('i', [10, 0, 0, 40])
print("Original Array:")
for element in arr: 
    print(element, end=" ")  # Added end=" " for better formatting
print()  # New line

print("Manipulated Array:")
move_zeroes_bubble_array(arr)
for element in arr:  # Print the modified array
    print(element, end=" ")
print()

File: Algorithms/Sorting-Algorithms/Sorting-Problems/move_zeroes/Java/MoveZeroes.java
// Problem Statement : Move Zeroes to the end of the array.....


import java.util.*;
import java.lang.*;
import java.io.*;

public class MoveZeroes{
	
	public static void moveZeroes(int[] arr){
		int n = arr.length;
		
		boolean flag = false;
		
		
		for(int i=0 ; i<n ; i++){
			for(int j=i+1 ; j<n-i-1 ; j++){
			
				// finding elements where j -> 0 and j+1 -> non-zero
				if(arr[j] == 0 && arr[j+1] != 0){
					int temp = arr[j];
					arr[j] = arr[j+1];
					arr[j+1] = temp;
					
					flag = true;
				}
			}
			
			if(flag == false){
				break;
			}
		}
		
		for (int val : arr){
			System.out.print(val + " ");
		}
		

	}
	
	public static void main(String[] args){
		int[] arr = {4,6,0,1,0,2,3};
		
		moveZeroes(arr);
		
		System.out.println();
	}
}
